src/client.rs

```diff
@@ -153,9 +161,13 @@ impl ClientFlow {
                     }
                 }
                 Response::Data(data) => break Some(ClientFlowEvent::DataReceived { data }),
-                Response::CommandContinuationRequest(_) => {
-                    self.send_command_state.continue_command();
-                    break None;
+                Response::CommandContinuationRequest(continuation) => {
+                    if self.send_command_state.command_in_progress().is_some() {
+                        self.send_command_state.continue_command();
```

Comment:
How about `continue_command` returning a `bool` instead of checking `command_in_progress`?